randovania/gui/preset_settings/item_pool_tab.py

```diff
@@ -230,13 +230,26 @@ def _on_update_maximum_starting(self, value: int):
 
     def _create_categories_boxes(self, pickup_database: PickupDatabase, size_policy):
         self._boxes_for_category = {}
+        all_categories = list(pickup_database.pickup_categories.values())
 
         categories = set()
         for standard_pickup in pickup_database.standard_pickups.values():
             if not standard_pickup.hide_from_gui:
                 categories.add(standard_pickup.pickup_category)
 
-        all_categories = list(pickup_database.pickup_categories.values())
+        # FIXME: this really shouldn't be done this way. This is just temporary to have ammo in their own category.
+        broad_to_category = {
+            "beam_related": "beam",
+            "morph_ball_related": "morph_ball",
+            "missile_related": "missile",
+        }
+        for ammo_pickup in pickup_database.ammo_pickups.values():
+            if ammo_pickup.broad_category.name in broad_to_category:
+                proper_name = broad_to_category[ammo_pickup.broad_category.name]
+                categories.add(next(filter(lambda p: p.name == proper_name, all_categories)))
```

Comment:
Is this part really needed? It's only executed for one of `beam_related`, `morph_ball_related` or `missile_related`. So, you want to add one of those for AM2R without a main item in any of the mentioned categories?
Otherwise this only adds the only existing one.

It looks more like you want to just skip for `beam_related`, `morph_ball_related` or `missile_related`

Reply:
> So, you want to add one of those for AM2R without a main item in any of the mentioned categories?

Not quite, this is more needed as a hack to make the other games not regress. Otherwise on echoes for example, you suddenly get new empty "beam related" or "morph ball related" categories.

Reply:
You translate `beam related` to `beam` and then do `categories.add(...)`. `categories` is a set and it is just adding `beam` again, which is doing nothing in a set. It was already added by code which is above (and not shown) in the snippet. That's why I said, that I how I understand the code, you just want to skip it in a sense of `continue` when the condition is true instead of adding stuff.

Reply:
Ah, completely missed that it's a set.
For am2r, we can indeed just make it a continue. There's no case yet known to me where we need that stuff, and if some future case arrives, then dunc's rewrite hopefully happened by then.